erpnext/public/js/utils.js

```diff
@@ -568,7 +568,7 @@ erpnext.utils.select_alternate_items = function (opts) {
 					qty = row.qty;
 				}
 				row[item_field] = d.alternate_item;
-				frappe.model.set_value(row.doctype, row.name, "qty", qty);
+				frappe.model.set_value(row.doctype, row.name, "required_qty", qty);
```

Comment:
The fieldname is "required_qty" only for "Work Order Item", not for "BOM Item" and "Stock Entry Item". Change accordingly.

Also, always send the pull request to develop branch, will backport it to v15-hotfix.